pkg/pgmodel/querier/query_tools.go

```diff
@@ -21,37 +23,38 @@ type queryTools struct {
 
 // getMetricTableName gets the table name for a specific metric from internal
 // cache. If not found, fetches it from the database and updates the cache.
-func (tools *queryTools) getMetricTableName(metricName string, isExemplarQuery bool) (string, error) {
-	tableFetchQuery := getSampleMetricTableSQL
+func (tools *queryTools) getMetricTableName(metricSchema, metricName string, isExemplarQuery bool) (model.MetricInfo, error) {
+	metricInfo, err := tools.metricTableNames.Get(metricSchema, metricName, isExemplarQuery)
+	if err == nil {
+		return metricInfo, nil
+	}
+	if err != errors.ErrEntryNotFound {
+		return model.MetricInfo{}, fmt.Errorf("fetchign metric info from cache: %w", err)
+	}
+
 	if isExemplarQuery {
 		// The incoming query is for exemplar data. Let's change our parameters
 		// so that the operations with the database and cache is focused towards
 		// exemplars.
-		tableFetchQuery = getExemplarMetricTableSQL
+		tableName, err := queryExemplarMetricTableName(tools.conn, metricName)
+		if err != nil {
+			return model.MetricInfo{}, fmt.Errorf("query exemplar metric table: %w", err)
+		}
```

Comment:
we are never setting the cache in this branch

Reply:
Why would we set the cache if we had any error in querying exemplar table? same goes for the line `metricInfo, err = querySampleMetricTableName(tools.conn, metricSchema, metricName)` below.